src/shared/modules/__tests__/helpers.test.ts

```diff
@@ -364,4 +494,95 @@ describe('modules.helpers', () => {
             );
         });
     });
+
+    describe('queue helpers', () => {
+        it('add common control to the end of queue', () => {
+            const itemsStateAndParams: ItemsStateAndParams = {
+                [META_KEY]: {
+                    queue: [
+                        {id: DEFAULT_CONTROL_ID},
+                        {id: DEFAULT_CONTROL_ID_2, groupItemId: DEFAULT_GROUP_ITEM_ID},
+                    ],
+                    version: 2,
+                },
+            };
+
+            const config = getMockedConfig({
+                items: [
+                    getMockedControlItem({}),
+                    getMockedGroupControlItem({id: DEFAULT_CONTROL_ID_2}),
+                ],
+            });
+
+            expect({
+                queue: [
+                    {id: DEFAULT_CONTROL_ID_2, groupItemId: DEFAULT_GROUP_ITEM_ID},
+                    {id: DEFAULT_CONTROL_ID},
+                ],
+                version: 2,
+            }).toEqual(addToQueue({id: DEFAULT_CONTROL_ID, config, itemsStateAndParams}));
+        });
+
+        it('remove unused items from queue after adding new item', () => {
```

Comment:
Can u add case where we have 1 items with 2 different groups in queue but one group id doesn't exist in the `config`? And we add something to queue - it should delete this group id from queue